sim/data_provider.py

```diff
@@ -0,0 +1,37 @@
+# pylint: disable=too-many-arguments, too-many-locals
+
+from abc import ABC, abstractmethod, abstractproperty
+from typing import Union
+import xarray as xr
+
+
+class IDataProvider(ABC):
+    """Interface for all data providers.
+    A DataProvider serves as the backing data for simulation
+    """
+
+    @abstractproperty
```

Comment:
abstractproperty is deprecated according to https://docs.python.org/3/library/abc.html#abc.abstractproperty